tests/runtime/conftest.py

```diff
@@ -50,6 +50,11 @@
 log = alog.use_channel("TEST-CONFTEST")
 
 
+def get_open_port():
+    """Non-fixture function to get an open port"""
+    return _open_port()
+
+
 @pytest.fixture
 def open_port():
```

Comment:
@gabe-l-hart, there's a slightly better approach to getting a next available port.  
In the caikit-nlp-client library that @dtrifiro and I have developed we execute this function:

```python
def get_random_port():
    with closing(socket.socket(socket.AF_INET, socket.SOCK_STREAM)) as s:
        s.bind(("", 0))
        s.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
        port = s.getsockname()[1]
        return port
```
Basically it returns a random open port from the whole range of ports on the host. We have not seen any problems with it running into any collisions and I have seen this technique work very well in other languages. Here's the [link to the original](https://github.com/z103cb/caikit-nlp-client/blob/c797414e68ee3723636ee84375eda01d8c1ee318/tests/fixtures/utils.py#L13C1-L18C20) to the code. We used this [SO Post](https://stackoverflow.com/questions/1365265/on-localhost-how-do-i-pick-a-free-port-number) as the basis of the implementation.